app/src/main/java/agency/five/codebase/android/movieapp/ui/component/MovieCategoryLabel.kt

```diff
@@ -0,0 +1,102 @@
+package agency.five.codebase.android.movieapp.ui.component
+
+import agency.five.codebase.android.movieapp.R
+import agency.five.codebase.android.movieapp.ui.theme.LocalSpacing
+import agency.five.codebase.android.movieapp.ui.theme.Typography
+import androidx.annotation.StringRes
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.Divider
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+
+data class MovieCategoryLabelViewState(
+    val itemId: Int,
+    val isSelected: Boolean,
+    val categoryText: MovieCategoryLabelTextViewState
+)
+
+sealed class MovieCategoryLabelTextViewState {
+    class LabelTextFromString(val text: String) : MovieCategoryLabelTextViewState()
+    class LabelTextFromResource(@StringRes val textRes: Int) : MovieCategoryLabelTextViewState()
+}
+
+@Composable
+fun MovieCategoryLabel(
+    modifier: Modifier = Modifier,
+    onClick: () -> Unit,
+    labelViewState: MovieCategoryLabelViewState
+) {
+    Column(
+        modifier = modifier
```

Comment:
You might have noticed that underline is much longer than expected (it's width should be the same as the width of text.
Look into this function : `Modifier.width(intrinsicSize: IntrinsicSize)` and try adding it here.